x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -250,6 +251,37 @@ export class FleetPlugin
           },
         },
       });
+
+      deps.features.registerKibanaFeature({
+        id: INTEGRATIONS_PLUGIN_ID,
+        name: 'Integrations',
+        category: DEFAULT_APP_CATEGORIES.management,
+        app: [INTEGRATIONS_PLUGIN_ID, 'kibana'],
+        catalogue: ['fleet'],
+        // Integrations app shares the same privileges as Fleet
+        privileges: {
+          all: {
+            api: [`${PLUGIN_ID}-read`, `${PLUGIN_ID}-all`],
+            app: [INTEGRATIONS_PLUGIN_ID, 'kibana'],
+            catalogue: ['fleet'],
+            savedObject: {
+              all: allSavedObjectTypes,
+              read: [],
+            },
+            ui: ['show', 'read', 'write'],
+          },
+          read: {
+            api: [`${PLUGIN_ID}-read`],
+            app: [INTEGRATIONS_PLUGIN_ID, 'kibana'],
+            catalogue: ['fleet'], // TODO: check if this is actually available to read user
+            savedObject: {
+              all: [],
+              read: allSavedObjectTypes,
+            },
+            ui: ['show', 'read'],
+          },
+        },
+      });
```

Comment:
I don't think you want a new Kibana feature registered. Based on a comment from @jen-huang, we expect that users of the integrations application will also require privileges to the existing fleet app. Similarly, it's expected that users of the fleet app will be able to use this integrations app.

If that is all true, then you should just need to declare your integrations app within the `app` array of the existing fleet feature above. The UI Capabilities (`ui`) and API access guards (`api`) will be scoped to the `fleet` feature instead of the new `integrations` feature, but that should be tolerable if I understand your requirements.

^^ Let me know if any of that is wrong and we'll work out what we need to do

Reply:
Makes perfect sense to me, thank you. I think this should work as we're expecting here. I just pushed up a commit addressing this, and will verify in CI.